### ISSUE TITLE
`cargo contract info --all` stops working on Chopsticks after some interactions

### ISSUE THREAD

**Issue by cmichi** [paritytech/cargo-contract#1475]
## How to reproduce

(I've patched my `cargo-contract` to use `ink_metadata` from ink! `master`.)

### Step 1

Start `substrate-contracts-node`. It'll run the RPC server on port 9944.

### Step 2

Deploy `flipper`:

```
cd ink/integration-tests/flipper/
cargo contract build --release
cargo contract instantiate --suri //Alice --args true -x 
```

Check that it's there:
```
$ cargo contract info --all --url=ws://localhost:9944
5D4zzvxGZq4wx3SYuaDomSMSAS1h1Knm35gz62UAk4Tqscey
```

### Step 3

Create a fork of the chain from the `substrate-contracts-node`, using [Chopsticks](https://github.com/AcalaNetwork/chopsticks/).

```
$ cat dev.yml
endpoint: ws://127.0.0.1:9944
mock-signature-host: true
block: 1
db: ./db.sqlite

import-storage:
  Sudo:
    Key: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY # Alice

$ npx @acala-network/chopsticks@latest --config=dev.yml 
```

The Chopsticks RPC will be run on port 8000. Check that the contract appears there:
```
$ cargo contract info --all --url=ws://localhost:9944
5D4zzvxGZq4wx3SYuaDomSMSAS1h1Knm35gz62UAk4Tqscey
```

### Step 4

Deploy another `flipper` to the branched of Chopsticks chain:

```
cd flipper/
cargo contract instantiate --suri //Alice --args true -x --salt 10 --url=ws://localhost:8000
```

If you now execute `cargo contract info --all --url=ws://localhost:8000` it won't show any contracts at all. The contracts still exist on-chain and can be interacted with. Just that `info --all` doesn't show them anymore.

I suspect something with smoldot (which is used by Chopsticks) not working correctly with `subxt`.

**Comment by smiasojed** [comment]
The issue is that after creating another block with a contract in chopstick, when we query storage for this block, it does not have the contract entry. However, this newly created contract entry is visible when we query the block at which we forked with chopstick.

Another issue is that the RPC method "state_getKeysPaged" does not return any value when we add a strict key as a method parameter. Even though the key is present in the block, we will receive an empty response.

**Comment by smiasojed** [comment]
The issue is related to the state_getKeysPaged RPC call, which seems to not work correctly. However, when you build chopstick from the master branch, the issue seems to be resolved.

**Comment by cmichi** [comment]
Thanks for the investigation, I'll close the issue then and hope they'll do a release soon.

**Comment by smiasojed** [comment]
> Thanks for the investigation, I'll close the issue then and hope they'll do a release soon.

Fixed in `Chopsticks` release 0.9.8
